update-solana-dependencies.sh

```diff
@@ -40,5 +40,5 @@ crates=(
 
 set -x
 for crate in "${crates[@]}"; do
-  sed -i'' -e "s#\(${crate} = \"\)\(=\?\).*\(\"\)#\1\2$solana_ver\3#g" "${tomls[@]}"
+  sed -E -i'' -e "s#(${crate} = \")(=?).*#\1\2${solana_ver}\"#" "${tomls[@]}"
```

Comment:
I found this change upstream in `solana-program-library` and it allows this script to work on Macs.